### ISSUE TITLE
error: 'ttir.scatter' op Batching currently not supported

### ISSUE THREAD

**Issue by upantelicTT** [tenstorrent/tt-mlir#4139]
### Describe the bug

When using jnp.take_along_axis inside a JIT-compiled loss function and computing gradients via jax.grad, the backward pass triggers a stablehlo.scatter operation. This is due to how autodiff handles the gradient of take_along_axis. However, the ttir.scatter op generated during lowering is currently not supported with batching, and compilation fails.
```python

# Loss that includes jnp.take_along_axis
def loss_fn(logits, labels):
    log_probs = jax.nn.log_softmax(logits)
    picked = jnp.take_along_axis(log_probs, labels[:, None], axis=-1)
    return -picked.mean()

# Add grad to trigger scatter on backward
@nnx.jit
def failing_backward(logits, labels):
    grad_fn = jax.grad(loss_fn)
    return grad_fn(logits, labels)

# Input
logits = jnp.array([
    [0.1, 0.5, 0.4],
    [0.2, 0.2, 0.6]
])
labels = jnp.array([1, 2])

# This should trigger scatter on TT and fail
print(failing_backward(logits, labels))

``` 

### Steps to reproduce the issue

link for stablehlo -> https://gist.github.com/upantelicTT/57c24d0ac1f2dc2c3d9f7fcba2ea2394

error: 'ttir.scatter' op Batching currently not supported

%1 = "stablehlo.scatter"(%0, %arg0, %arg1) <{scatter_dimension_numbers = #stablehlo.scatter<inserted_window_dims = [1], input_batching_dims = [0], scatter_indices_batching_dims = [0], scatter_dims_to_operand_dims = [1], index_vector_dim = 2>}> ({
    ^bb0(%arg2: tensor<f32>, %arg3: tensor<f32>):
      %2 = stablehlo.add %arg2, %arg3 : tensor<f32>
      stablehlo.return %2 : tensor<f32>
    })

**Comment by sdjordjevicTT** [comment]
Currently, TT-Metal works on a proper implementation of the scatter op. Here we can check the status:
https://github.com/tenstorrent/tt-metal/issues/24616
